https://issues.redhat.com/browse/ACM-13640 import managedhub in hosted mode (#6898)
open-cluster-management/rhacm-docs: global_hub/global_hub_import_hub_hosted.adoc

[#global-hub-importing-managed-hub-in-hosted-mode]

> = Importing a managed hub cluster in the hosted mode (Technology Preview)

Reviewer: we should add a per-condition since this feature required the global hub control plane have `local-cluster` enabled since the `local-cluster` will be used as the hosting cluster.

To import existing hub cluster which has `local-cluster` enabled, you must use hosted mode to import it.

Set the annotation `global-hub.open-cluster-management.io/import-cluster-in-hosted=true` to globalhub cr will importing all the new managed hub clusters in hosted mode.

With this annotation, globalhub controller will do the following things:

> 1. Import all the new managedhub clusters in hosted mode.

Reviewer: change managedhub clusters -> managed hub clusters

> 2. Disable addons `applicationManager` `certPolicyController` `policyController` in all new managedhub clusters namespaces.

Reviewer: change managedhub clusters -> managed hub clusters

> 3. Change addons `work-manager`,`cluster-proxy`,`managed-serviceaccount` namespace to `open-cluster-management-global-hub-agent-addon`

Reviewer: will this impact the existing clusters? or it only impact the new managed hub clusters?

Author: all managedhub related addons will be impacted, changed.

[#global-hub-importing-managed-hub-in-hosted-mode]

> = Importing a managed hub cluster in the hosted mode (Technology Preview)

Reviewer 2: What do you guys think about creating a "why would I do this?"  or "benefit of doing this?" statement here, right after the title? Something like the following: 

```suggestion
= Importing a managed hub cluster in the hosted mode (Technology Preview)

Import a managed hub cluster in the hosted mode to import any managed hub cluster you want and change the `addons` in the namespace. 
```

Author: added the line

> 

Reviewer 2: ```suggestion
[#global-hub-importing-prereq]
```

Author: done

== Prerequisites

> - You must enable `local-cluster` in global hub cluster.

Reviewer 2: ```suggestion
- Enable `local-cluster` in the {global-hub} cluster. 
```

Author: done

> To import existing hub cluster which has `local-cluster` enabled, you must use hosted mode to import it.

Reviewer 2: ```suggestion
To import existing hub cluster which has `local-cluster` enabled, complete the following steps: 
```

Author: removed this line

> Set the annotation `global-hub.open-cluster-management.io/import-cluster-in-hosted=true` to globalhub cr will importing all the new managed hub clusters in hosted mode.

Reviewer 2: ```suggestion
. Use the hosted mode to import the existing hub cluster. 
. Set the annotation `global-hub.open-cluster-management.io/import-cluster-in-hosted=true` to `global hub`.  
. Use the custom resource to import all the new managed hub clusters in the hosted mode.
```

> With this annotation, globalhub controller will do the following things:

Reviewer 2: ```suggestion
With this annotation, you can use the {global-hub} controller to do the following features: 
```

Author: The logic is not like this, I changed it, and please review again.

> 1. Import all the new managed hub clusters in hosted mode.

Reviewer 2: ```suggestion
- Import all the new managed hub clusters in hosted mode.
```

Author: done

> 2. Disable addons `applicationManager` `certPolicyController` `policyController` in all new managed hub clusters namespaces.

Reviewer 2: ```suggestion
- Disable the following `addons` in the new managed hub clusters namespaces: `applicationManager`, `certPolicyController`, and `policyController`.  
```

Author: done

> 3. Change addons `work-manager`,`cluster-proxy`,`managed-serviceaccount` namespace to `open-cluster-management-global-hub-agent-addon`(All managed hub clusters related addons will be updated to new namespace)

Reviewer 2: ```suggestion
- Change the managed hub clusters `addons` that are related to the new namespace, including:  `work-manager`,`cluster-proxy`, and `managed-serviceaccount`. Change these namespaces to `open-cluster-management-global-hub-agent-addon`. 
```

Author: done

> Notes: In this mode, we support import managed hub clusters using kubeconfig or api token. RHACM will use this kubeconfig or api token to generate the `auto-import-secret`, which used to connect to managed hub cluster, you need to make sure this kubeconfig or api token is always usable, if it become expired, you need to regenerate the `auto-import-secret` in managed hub namespace

Reviewer 2: ```suggestion
*Notes:*
- In the hosted mode, you only get support for import managed hub clusters that use `kubeconfig` or API tokens. {acm} uses this `kubeconfig` or API token to generate the `auto-import-secret`, which connects to the managed hub cluster.
- Make sure this `kubeconfig` or API token is always usable. If it expires,  regenerate the `auto-import-secret` in the managed hub namespace, `oc create secret generic auto-import-secret --from-file=kubeconfig=./managedClusterKubeconfig -n cluster1`
```

Author: done

> `oc create secret generic auto-import-secret --from-file=kubeconfig=./managedClusterKubeconfig -n cluster1`

Reviewer 2: remove this line if you apply my previous comment.

Author: done

[#global-hub-importing-managed-hub-in-hosted-mode]
= Importing a managed hub cluster in the hosted mode (Technology Preview)

> Import a managed hub cluster in the hosted mode to import any managed hub clusters you want to have `local-cluster` enabled. 

Reviewer 2: ```suggestion
Import a managed hub cluster in the hosted mode to import any managed hub clusters where you want to have `local-cluster` enabled. 
```

Author: done

[#global-hub-importing-prereq]
== Prerequisites
- Enable `local-cluster` in the {global-hub} cluster.

To import existing hub cluster which has `local-cluster` enabled, you must use the hosted mode to import the existing hub cluster. Complete the following steps: 

> . Set the annotation `global-hub.open-cluster-management.io/import-cluster-in-hosted=true` to globalhub cr.

Reviewer 2: ```suggestion

. Set the annotation `global-hub.open-cluster-management.io/import-cluster-in-hosted=true` to globalhub cr.
```

Reviewer 2: Basically, we need an empty line on line 11 in between `Complete the following steps` and `Set the annotation...`

So it'd be:

Complete the following steps:

1. Set the annotation...

Author: done

> With this annotation, {global-hub} will do the following things:

Reviewer 2: ```suggestion
With this annotation, {global-hub} does the following actions:
```

Author: done
